gallery/src/components/NxTable/NxTableSortableExample.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, {useState} from 'react';
+
+import {
+  NxTable,
+  NxTableCell,
+  NxTableHead,
+  NxTableRow
+} from '@sonatype/react-shared-components';
+
+const initialState = ['A', '1', 'B', '2', 'C', '3', 'D', '4'];
+
+const NxTableSortableExample = () => {
+  const [rows, setRows] = useState(initialState);
+  const [sortDir, setSortDir] = useState<'desc' | 'asc' | null>(null);
+
+  function sort() {
+    if (sortDir === null) {
```

Comment:
I think the logic of the null -> asc -> desc -> null -> ... ordering of the sort button cycling should be encapuslated into the RSC lib in some way.  Seems like a candidate for a "stateHelper" as patterned in `NxTextInput`

Reply:
It may depend on the UI that uses the table. In one case you might want one column to always show the sort indicator and have the other columns indicated they are unsorted but sortable. In another case you might want the user to be able to clear all column sorting. Maybe we want to think about codifying a particular kind of UX though? We might want to think about adopting a library like react-table to provide that stateful behavior while keeping things pretty customizable. I though it better to avoid dealing with sorting logic for now. We can always come back and introduce something later.

Reply:
> In one case you might want one column to always show the sort indicator and have the other columns indicated they are unsorted but sortable.

I don't follow what you're describing here.

> We might want to think about adopting a library like react-table to provide that stateful behavior while keeping things pretty customizable. I though it better to avoid dealing with sorting logic for now. We can always come back and introduce something later.

To clarify, I wasn't suggesting that we encapsulate the actual sorting of the data, only the standard sequence of sort button states as you click on it

Reply:
I see now, though I'm not convinced it's going to be all that useful. If you're ok with shipping this without I'd prefer to.

Reply:
> I see now, though I'm not convinced it's going to be all that useful. If you're ok with shipping this without I'd prefer to.

What do other think? @sonatype/shared-components-group

Reply:
> If you're ok with shipping this without I'd prefer to.

I'm fine with not shipping this in this PR, as long as we don't lose track of it. Maybe create a ticket to fllow up on this?